stdlib/LinearAlgebra/test/generic.jl

```diff
@@ -355,6 +355,10 @@ end
     @test [[1,2, [3,4]], 5.0, [6im, [7.0, 8.0]]] ≈ [[1,2, [3,4]], 5.0, [6im, [7.0, 8.0]]]
 end
 
+@testset "Issue 40128" begin
+               @test LinearAlgebra.det_bareiss(BigInt[1 2; 3 4]) == -2
```

Comment:
Fix the indenting here.  (4 spaces)

Reply:
Also check that the result is a `BigInt`

Reply:
Specifically, you should probably test something like `BigInt[1 big(2)^65+1; 3 4])` which will ensure that no `Float` based method would produce the right answer.

Reply:
You could try `det(BigInt[9 1 8 0; 0 0 8 7; 7 6 8 3; 2 9 7 7])::BigInt == -1`, since for `Int` this is off by about `1e-12`.

Reply:
> Specifically, you should probably test something like BigInt[1 big(2)^65+1; 3 4]) which will ensure that no Float based method would produce the right answer.

Actually the current algorithm gives the exact answer here:
```jl
julia> det(BigInt[1 big(2)^65+1; 3 4]) - (4 - 3*(big(2)^65+1))
0.0
```
(Remember that it uses `BigFloat`.)